radix-engine/src/engine/process.rs

```diff
@@ -603,24 +669,15 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
                     .track
                     .get_resource_manager(&resource_address)
                     .unwrap()
-                    .get_auth(&function, &args);
+                    .get_consuming_bucket_auth(arg);
                 Ok((Consumed(Some(ConsumedSNodeState::Bucket(bucket))), vec![method_auth.clone()]))
             }
             SNodeRef::BucketRef(bucket_id) => {
                 let bucket = self
                     .buckets
                     .remove(&bucket_id)
                     .ok_or(RuntimeError::BucketNotFound(bucket_id.clone()))?;
-                let resource_address = bucket.resource_address();
-                let method_auth = self
-                    .track
-                    .get_resource_manager(&resource_address)
-                    .unwrap()
-                    .get_auth(&function, &args);
-                Ok((
-                    Borrowed(BorrowedSNodeState::Bucket(bucket_id.clone(), bucket)),
-                    vec![method_auth.clone()],
-                ))
+                Ok((Borrowed(BorrowedSNodeState::Bucket(bucket_id.clone(), bucket)), vec![]))
```

Comment:
No, as you can see in all of the `BorrowedSNodeState` invariants, these are all owned objects. It's "borrowed" because it is exposed as borrowed through the `SNodeState` interface as something borrowed during execution. The borrowed states are then returned at the end of the call.

Reply:
Yep, understood the implementation. I was just noting a direction that we may consider. The limitation of manual borrow semantics is hard to maintain and prone to bugs.

Reply:
Ah, it's just difficult/impossible to do because we ARE implementing `borrow_mut` essentially. The problem comes in the Rust constraint that you should never more than one mutable borrow so this makes it very difficult to both pass a borrowed object and a borrowed track to a called process.